platform/android/java/lib/src/org/godotengine/godot/Godot.kt

```diff
@@ -650,6 +663,13 @@ class Godot(private val context: Context) : SensorEventListener {
 					}
 				}
 			}
+		} else {
+			val shouldGoImmersive = java.lang.Boolean.parseBoolean(GodotLib.getEditorSetting("interface/editor/android/immersive_mode"))
+
+			runOnUiThread {
+				if (shouldGoImmersive)
+					goImmersive()
+			}
```

Comment:
Let's move this logic to `GodotEditor` and have that class invoke `Godot::enableImmersive()` as needed.

There is one additional thing we need to consider; should the running project window also open in immersive mode if the editor is immersive?
- If yes, then the logic can be copied as is `GodotEditor`
- If no, then you need to provide a method that `GodotGame.kt` can override to declare whether it should be immersive. See [`GodotEditor#enableLongPressGestures()`](https://github.com/godotengine/godot/blob/97b8ad1af0f2b4a216f6f1263bef4fbc69e56c7b/platform/android/java/editor/src/main/java/org/godotengine/editor/GodotEditor.kt#L273) and [`GodotGame#enableLongPressGestures()`](https://github.com/godotengine/godot/blob/97b8ad1af0f2b4a216f6f1263bef4fbc69e56c7b/platform/android/java/editor/src/main/java/org/godotengine/editor/GodotGame.kt#L44) for example.

Reply:
I've added a separate option for the game. This option uses the `--use_immersive` functionality that already existed.